integration_test/support/schemas.exs

```diff
@@ -53,9 +53,20 @@ defmodule Ecto.Integration.Post do
     has_one :force_permalink, Ecto.Integration.Permalink, on_replace: :delete_if_exists
     has_one :update_permalink, Ecto.Integration.Permalink, foreign_key: :post_id, on_delete: :delete_all, on_replace: :update
     has_many :comments_authors, through: [:comments, :author]
+
+    has_many :v2_comments, Ecto.Integration.Comment, where: [lock_version: 2]
+    has_many :v2_comments_authors, through: [:v2_comments, :author]
+
     belongs_to :author, Ecto.Integration.User
     many_to_many :users, Ecto.Integration.User,
       join_through: "posts_users", on_delete: :delete_all, on_replace: :delete
+
+    # Tests a has through with a many to many and a has many
+    has_many :posts_through_users, through: [:users, :posts]
+
+    # Tests a has through with two many to many associations
+    has_many :posts_through_posts, through: [:users, :schema_posts]
```

Comment:
This is schema is already super large. Any chance we can write this test by moving this additional associations to the User schema instead? 🤔

Reply:
Sure I pivoted to queries on user